### PR TITLE
feat(rule): autofocus attribute should not be used

### PR DESCRIPTION
**Pull request by mohammedzamakhan**

### CODE REVIEW

File: test/templateNoAutofocusRule.spec.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { Rule } from '../src/templateNoAutofocusRule';
+
+const {
+  FAILURE_STRING,
+  metadata: { ruleName }
+} = Rule;
+
+describe(ruleName, () => {
+  describe('failure', () => {
+    it('should fail if async pipe is negated', () => {
```
Comment (mgechev) [inline test/templateNoAutofocusRule.spec.ts]: This should be updated

File: test/templateNoAutofocusRule.spec.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        }
+      `;
+      assertAnnotated({
+        message: FAILURE_STRING,
+        ruleName: ruleName,
+        source
+      });
+    });
+
+    it('should fail if async pipe is negated 1', () => {
```
Comment (mgechev) [inline test/templateNoAutofocusRule.spec.ts]: Message should be updated

File: test/templateNoAutofocusRule.spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      assertAnnotated({
+        message: FAILURE_STRING,
+        ruleName: ruleName,
+        source
+      });
+    });
+  });
+
+  describe('success', () => {
+    it('should succeed if async pipe is not negated', () => {
```
Comment (mgechev) [inline test/templateNoAutofocusRule.spec.ts]: Same

File: src/templateNoAutofocusRule.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import { SourceFile } from 'typescript/lib/typescript';
+import { NgWalker } from './angular/ngWalker';
+import { BasicTemplateAstVisitor } from './angular/templates/basicTemplateAstVisitor';
+
+export class Rule extends Rules.AbstractRule {
+  static readonly metadata: IRuleMetadata = {
+    description: 'Ensure that autofocus property is not used',
+    options: null,
+    optionsDescription: 'Not configurable.',
+    rationale: 'autofocus attribute reduces usability and accessibility for users. (w3c warning)',
```
Comment (mgechev) [inline src/templateNoAutofocusRule.ts]: you can drop `. (w3c warning)`

**Comment by mgechev** [comment]
Left few minor comments. Would you share details about `autofocus` considered as a bad practice?

**Comment by mohammedzamakhan** [comment]
@mgechev you can find the warning here https://w3c.github.io/html/sec-forms.html#autofocusing-a-form-control-the-autofocus-attribute

**Comment by rgant** [comment]
Seems like W3C has removed the warning about autofocus now. I don't see anything referencing autofocus in the July 18, 2019 version of the document. I would like to review the discussion if I could find it.

**Pull request merged**